### PR TITLE
feature(nyz): add PPOF new interface support

### PR DESCRIPTION
**Pull request by PaParaZz1**
## Description
```python
from ding.bonus import PPOF

agent = PPOF(env='lunarlander_discrete', exp_name='lunarlander_discrete_demo')
agent.train(step=int(1e5))
agent.deploy(enable_save_replay=True)
```

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/bonus/ppof.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+class PPOF:
+    supported_env_list = [
+        'lunarlander_discrete',
+        'lunarlander_continuous',
+        'rocket_landing',
+        'drone_fly',
+        'hybrid_moving',
+    ]
+
```
Comment (puyuan1996) [inline ding/bonus/ppof.py:26]: 添加 bonus 路径下的文件说明？简介这里的ppof与policy/下面的ppof的异同？

File: dizoo/rocket/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Install
+
+```shell
+pip install git+https://github.com/nighood/rocket-recycling@master#egg=rocket_recycling
```
[the remark this answers is not in the document]
Comment (PaParaZz1) [reply dizoo/rocket/README.md:4]: 这个暂时不用吧，等稳定版

File: ding/bonus/ppof.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    def collect_demo_data(
+            self,
+            env_num: int = 8,
+            ckpt_path: Optional[str] = None,
+            save_data_path: Optional[str] = None,
+            n_sample: Optional[int] = None,
+            n_episode: Optional[int] = None,
+            debug: bool = False
+    ) -> None:
```
[the remark this answers is not in the document]
Comment (PaParaZz1) [reply ding/bonus/ppof.py:125]: ok

File: ding/bonus/ppof.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        step = 0
+        obs = env.reset()
+        while True:
+            action = forward_fn(obs)
+            obs, rew, done, info = env.step(action)
+            return_ += rew
+            step += 1
+            if done:
+                break
+        logging.info(f'PPOF deploy is finished, final epsiode return with {step} steps is: {return_}')
```
Comment (puyuan1996) [inline ding/bonus/ppof.py:115]: epsiode -> episode

File: ding/envs/env_manager/subprocess_env_manager.py [hunk 5]
```diff
@@ -834,7 +834,9 @@ def step(self, actions: List[tnp.ndarray]) -> List[tnp.ndarray]:
             - timesteps (:obj:`List[tnp.ndarray]`): Each timestep is a tnp.array with observation, reward, done, \
                 info, env_id.
         """
-        actions = {env_id: a for env_id, a in zip(self.ready_obs_id, actions)}
+        split_action = tnp.split(actions, actions.shape[0])
```
[the remark this answers is not in the document]
Comment (PaParaZz1) [reply ding/envs/env_manager/subprocess_env_manager.py:837]: 已添加注释

File: ding/bonus/ppof.py (hunk 4 above)
Comment (PaParaZz1) [reply ding/bonus/ppof.py:115]: fixed

**Pull request merged**